### ISSUE TITLE
Thank you for work! A few questions

### ISSUE THREAD

**Issue by CubicQubit** [BorgwardtLab/Set_Functions_for_Time_Series#8]
Hi there,

Thank you for both the `medical_ts_dataset` repo and `this` repo. Very reproducible work, which is super exciting for healthcare research.

I was wondering if you can help me reconcile your work and this paper: [https://arxiv.org/abs/2101.10318](https://arxiv.org/abs/2101.10318)

Both has a MIMIC3 table using AUROC as the point of comparison. However, the numbers and ordering of methods are quite different.  I understand that this work followed `mimic3-benchmarks` train-test-val splits and IP-net/mTAND followed a different data split.  However, do you suppose that would contribute to such a difference , or could it also be some sort of data-processing/normalization process that are different between the two works?

Mainly, I'm also interested in the authors take on the current research progress for irregular time-series classification. There have been many papers demonstrating superior performance over GRU-D yet in this work, GRU-D is quite strong if not the best for MIMIC3, in addition to being very strong for Physionet 2012 and 2019. (I also followed up with the issue between different masking for x_t and m_t, and setting them to be the same doesn't downgrade performance that much, which implies that GRU-D is still strong). Is it still SOTA?

**Comment by ExpectationMax** [comment]
Hi CubicQubit,

Thank you for your patience.  I don't really think this discussion is relevant to the software repo here. Feel free to reach out to me personally if you want to have a detailed discussion on some of the points you raised. Here are my two cents to why this might be the case:

 1. It looks like the splits from the mimic3-benchmarks could have some sort of distribution shift. We did not investigate this issue further, yet the high degree of regularisation applied in the mimic3-benchmarks paper for the tasks we considered seems to gives some supporting evidence. We also mentioned this in our paper, yet understand that this is not conclusive evidence of course. Further, many the performance of our models on the validation set did not seem indicative of test set performance. And yes there are for sure differences in how the data was process between both works, making it hard to conclusively compare them.
 2. In our study we performed a very extensive and as fair as possible hyperparameter search, where each model is evaluated based on the same selection criteria. It could also be that GRU-D is much more robust to the selection of hyperparameters, whereas others are not. Due to limited compute budget, we evaluated the models on a limited number of random samples from predefined ranges. If methods are less robust to the hyperparameters and require more detailed tuning, it could be that we missed the perfect hyperparameters in out study leading to worse performance.
 3. Generally, comparability of studies in this realm is very difficult as only very few publications use predefined tasks and datasets which leads to the issues you described above. In generally, I cannot say if GRU-D is SOTA, in our study it most certainly resembled a strong competitor.

Hope that helps!

Cheers,
Max
